app/src/test/java/ch/hikemate/app/authentication/AuthViewModelTest.kt

```diff
@@ -189,28 +220,88 @@ class AuthViewModelTest {
       runTest {
         setupSignedOutUser()
 
-        // Simulate a successful email and password account creation by invoking the onSuccess
-        // callback
+        // Simulate successful creation of account
         doAnswer { invocation ->
               val onSuccess = invocation.getArgument<(FirebaseUser?) -> Unit>(0)
-              onSuccess(mockFirebaseUser) // Call the success callback with a mock user
+              onSuccess(mockFirebaseUser)
               null
             }
-            .`when`(mockRepository)
+            .whenever(mockRepository)
             .createAccountWithEmailAndPassword(any(), any(), any(), any())
 
-        // Verify that currentUser is initially null
-        assertNull(viewModel.currentUser.first())
+        doAnswer { invocation ->
+              val onSuccess = invocation.getArgument<(Profile) -> Unit>(1)
+              onSuccess(mockProfileUser) // Trigger success callback with mock profile
+              null
+            }
+            .whenever(mockProfile)
+            .createProfile(any(), any(), any())
 
-        viewModel.createAccountWithEmailAndPassword(
-            "mock@example.com", "password", {}, { fail("Error callback should not be called") })
+        // Create ArgumentCaptor for profile update request
+        val profileUpdateRequest = ArgumentCaptor.forClass(UserProfileChangeRequest::class.java)
 
-        val currentUser = viewModel.currentUser.first() // Get the first (current) value of the flow
+        // Mock user profile update to complete successfully
 
-        // Verify call to the repository
-        verify(mockRepository).createAccountWithEmailAndPassword(any(), any(), any(), any())
+        `when`(mockFirebaseAuth.createUserWithEmailAndPassword(any(), any()))
+            .thenReturn(mockAuthResult)
+        `when`(mockFirebaseUser.updateProfile(profileUpdateRequest.capture())).thenReturn(mockTask)
+        `when`(mockTask.isSuccessful).thenReturn(true)
 
-        assertEquals(mockFirebaseUser, currentUser)
+        `when`(mockTask.addOnCompleteListener(any())).thenAnswer { invocation ->
+          val listener = invocation.getArgument<OnCompleteListener<Void>>(0)
+          listener.onComplete(mockTask)
+          mockTask
+        }
+        // Mock task completion
+        `when`(mockTask.addOnCompleteListener(any())).thenAnswer { invocation ->
+          val listener = invocation.getArgument<OnCompleteListener<Void>>(0)
+          listener.onComplete(mockTask)
+          mockTask
+        }
+        `when`(mockAuthResult.addOnCompleteListener(any())).thenAnswer { invocation ->
+          val listener = invocation.getArgument<OnCompleteListener<Void>>(0)
+          listener.onComplete(mockTask)
+          mockTask
+        }
```

Comment:
Warning: I think you did some copy/paste spaghetti here :)

Reply:
Done